rest/src/main/scala/io/udash/rest/DefaultRestClient.scala

```diff
@@ -0,0 +1,70 @@
+package io.udash
+package rest
+
+import com.avsystem.commons._
+import com.avsystem.commons.annotation.explicitGenerics
+import com.avsystem.commons.meta.Mapping
+import com.softwaremill.sttp.Uri.QueryFragment.KeyValue
+import com.softwaremill.sttp.Uri.QueryFragmentEncoding
+import com.softwaremill.sttp._
+import io.udash.rest.DefaultSttpBackend.backend
+import io.udash.rest.raw._
+
+object DefaultRestClient {
+  @explicitGenerics def apply[RestApi: RawRest.AsRealRpc : RestMetadata](baseUri: String): RestApi =
+    RawRest.fromHandleRequest[RestApi](asHandleRequest(baseUri))
+
+  /**
+    * Creates a [[io.udash.rest.raw.RawRest.HandleRequest HandleRequest]] function which sends REST requests to
+    * a specified base URI using default HTTP client implementation (sttp).
+    */
+  def asHandleRequest(baseUri: String): RawRest.HandleRequest =
+    asHandleRequest(uri"$baseUri")
+
+  private def toSttpRequest(baseUri: Uri, request: RestRequest): Request[String, Nothing] = {
+    val uri = baseUri |>
+      (u => u.copy(path = u.path ++
+        request.parameters.path.map(_.value))) |>
+      (u => u.copy(queryFragments = u.queryFragments ++
+        request.parameters.query.iterator.map {
+          case (k, QueryValue(v)) => KeyValue(k, v, QueryFragmentEncoding.All, QueryFragmentEncoding.All)
+        }.toList
+      ))
+
+    val contentTypeHeader = request.body.mimeTypeOpt.map {
+      mimeType => (HeaderNames.ContentType, s"$mimeType;charset=utf-8")
+    }
+    val paramHeaders = request.parameters.headers.iterator.map {
+      case (n, HeaderValue(v)) => (n, v)
+    }.toList
+
+    sttp.copy[Id, String, Nothing](
+      method = Method(request.method.name),
+      uri = uri,
+      headers = contentTypeHeader.toList ++ paramHeaders,
+      body = request.body.contentOpt.map(StringBody(_, "utf-8")).getOrElse(NoBody)
+    )
+  }
+
+  private def fromSttpResponse(sttpResp: Response[String]): RestResponse =
+    RestResponse(
+      sttpResp.code,
+      Mapping(
+        sttpResp.headers.iterator.map {
+          case (n, v) => (n, HeaderValue(v))
+        }.toList,
+        caseInsensitive = true
+      ),
+      sttpResp.contentType.fold(HttpBody.empty) { contentType =>
+        val mimeType = contentType.split(";", 2).head
+        HttpBody(sttpResp.body.fold(identity, identity), mimeType)
+      }
+    )
+
+  private def asHandleRequest(baseUri: Uri): RawRest.HandleRequest =
+    RawRest.safeHandle(request => {
+      val sttpReq = toSttpRequest(baseUri, request)
```

Comment:
Inline?

Reply:
Oh why? It looks nice now. Originally I wanted to expose it for reuse.

Reply:
To delay `toSttpRequest` until the callback, should not matter much though.

Reply:
Actually, since this request is reusable, my specific intention was to *not* delay it until callback. Why do the same job potentially multiple times? If it throws exception then `RawRest.safeHandle` will take care of this.

Reply:
Convinced after last remark, did not notice that.